Update code of conduct to clarify reporting and moderation

juliepagano opened the pull request:
This PR:
- Adds section headings to make it easier to quickly parse.
- Adds easy to find contact information for people who need to report an issue.
- Adds link to TSC moderation policies, which are relevant to the code of conduct.
- Moves attribution to the bottom of the page (it's important, but probably not the first thing someone's looking for when they come to the page).

This PR was prompted by the inclusivity WG being notified that someone was having trouble figuring out who to report an incident to. Hopefully this will make that information easier to find.

juliepagano commented on CODE_OF_CONDUCT.md:
```diff
@@ -39,3 +37,19 @@ CoC](http://www.rust-lang.org/conduct.html) as well as the
+reported by:
+
+* Emailing [report@nodejs.org](mailto:report@nodejs.org) (this will email all TSC members)
+* Contacting [individual TSC members](https://nodejs.org/en/foundation/tsc/#current-members-of-the-technical-steering-committee).
```
Is there a better place I can link people for contact info for the TSC or is the right thing?

Fishrock123 replied:
Uh, looks like that's on me, not yet, but: https://github.com/nodejs/TSC/pull/1

juliepagano replied:
I'll leave it alone for now. We can change it when that lands.

bnoordhuis replied:
Isn't an email to report@nodejs.org the way to go?  Individual TSC members may have different ideas on how to deal with an issue so contacting one individually may lead to a certain level of arbitrariness.

juliepagano replied:
This is just explicitly repeating the contact guidelines in the [moderation policy](https://github.com/nodejs/TSC/blob/master/Moderation-Policy.md#requesting-moderation).

bnoordhuis replied:
Okay, filed an issue for that: https://github.com/nodejs/TSC/issues/45

Fishrock123 commented:
Heya, could you make sure the commit messages are max 50 chars and the descriptions are max 72?

Also I think the latter 3 commits could probably be squashed into one.

Seems fine to me otherwise. The contact details definitely need to be easily accessible.

TheAlphaNerd commented:
LGTM

juliepagano commented:
@Fishrock123 Sorry bout that. Was doing this quickly in the github UI and the commits were a little more quick and dirty than they should be. Will pull it down, clean things up a bit, and update shortly.

juliepagano commented:
I squashed it all into one commit since it's a pretty small change and modified the commit message to match the contribution guidelines. 

Is "doc" the right subsystem to list in the first line or should it be something else? Happy to change it something else if there's a specific thing I should use for policy documentation like this.

Fishrock123 commented:
@juliepagano Nah, we usually just use `doc:` for all of that.

jasnell commented:
LGTM

bnoordhuis commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,8 +1,6 @@
 
-This Code of Conduct is adapted from [Rust's wonderful
-CoC](http://www.rust-lang.org/conduct.html) as well as the
-[Contributor Covenant v1.3.0](http://contributor-covenant.org/version/1/3/0/).
```
Why is this section removed?

TheAlphaNerd replied:
moved to the bottom

zkat commented:
LGTM

evanlucas commented:
LGTM

Fishrock123 commented:
Thanks, landed in https://github.com/nodejs/node/commit/1e146e7c26be9add426f3cf92aa8c3d977de96e1
